kvdb/sqlbase/db.go

```diff
@@ -311,16 +311,23 @@ func (db *db) executeTransaction(f func(tx walletdb.ReadWriteTx) error,
 			return dbErr
 		}
 
-		err = catchPanic(func() error { return f(tx) })
-		if err != nil {
+		fnErr := catchPanic(func() error { return f(tx) })
+		if fnErr != nil {
 			if rollbackErr := tx.Rollback(); rollbackErr != nil {
 				log.Errorf("Error rolling back tx: %v",
 					rollbackErr)
 			}
 
-			return err
+			dbErr := MapSQLError(fnErr)
```

Comment:
In the edge case that the `Rollback()` fails we would probably better off returning instead of retrying.